pkg/docker/config/config.go

```diff
@@ -152,6 +172,30 @@ func GetCredentials(sys *types.SystemContext, registry string) (types.DockerAuth
 		return *sys.DockerAuthConfig, nil
 	}
 
+	credConfigs, err := sysregistriesv2.GetRegistryCredentialHelpers(sys, registry)
+	if err != nil {
+		return types.DockerAuthConfig{}, err
+	}
+	if len(credConfigs) > 0 {
+		if auth, err := func(credConfigs []string, registry string) (types.DockerAuthConfig, error) {
+			var lastError error
+			for _, credConfig := range credConfigs {
+				dockerAuthConfig, err := getAuthFromCredHelper(credConfig, registry)
```

Comment:
(Absolutely non-blocking: `dockerAuthConfig` can be just `authConfig`, `credentials` or even `creds`.)